### DISCUSSION TITLE
What does 'drop_ratio_build' and 'drop_ratio_search' actually means for sparse vector index and search?

### DISCUSSION THREAD

**Discussion by wangyiran33** [milvus-io/milvus#31805] [Q&A and General discussion]
In [https://milvus.io/docs/sparse_vector.md](url) I saw 
```
How should I choose the drop_ratio_build and drop_ratio_search parameters?

The choice of drop_ratio_build and drop_ratio_search depends on the characteristics of your data and your requirements for search latency/throughput and accuracy.
```
But I still don't know what these two parameters mean and if I'd like high accuracy which value should I set?

**Comment by yiwangdr** [comment]
@zhengbuqian fyi

@wangyiran33 
drop_ratio_build and drop_ratio_search essentially specify the percentage of small values that should be ignored from the build/search vectors.

Thus, please use the value 0 for high accuracy.

**Comment by zhengbuqian** [comment]
Thanks @yiwangdr, that is true!

For example `drop_ratio_build = 0.3`, imagine during the index building process you collect all values of all sparse vectors in the index and sort them, the smaller 30% will not be added to the index, reducing the amount of computation during search. `drop_ratio_search` does the same but to the query vector. 

We do this based on the nature of the IP metric: the larger the value is, the more it contributes to the final score. Thus dropping the smaller values had limited impact on the accuracy while greatly improving search performance.

When drop ratio is set to 0 for both build and search, the search is exhaustive with recall being 100%. If your use case allows certain levels of inaccuracy, it is recommended to use a non zero drop ratio. Per some of our tests on the SPLADE embeddings on the MS-MARCO dataset, decreasing the recall from 100% to 93%(by tuning the drop ratio) helped the search QPS to improve `>4x`(SPARSE_INVERTED_INDEX) or `>20x`(SPARSE_WAND).

**Comment by wangyiran33** [comment]
I totally understand now, thank you all! @zhengbuqian @yiwangdr

**Reply by zhengbuqian** [reply]
FYI I made a typo, and have edited my reply:

From "When drop ratio is set to 1 for both build and search, the search is exhaustive with recall being 100%."

To      "When drop ratio is set to 0 for both build and search, the search is exhaustive with recall being 100%."
